### PR TITLE
Plugin build updates

### LINKED ISSUE #90: Enable passing in goflags for plugin builds
**Is your feature request related to a problem? Please describe.**
When I build in an orphaned git submodule, I get an error,

```
building local repository at ./artifacts/plugins, v0.1.0, [all]
ðŸ¦ - building plugin at path "cmd/plugin/feature"
ðŸ¦ - running godep path "cmd/plugin/feature"
ðŸ¦$ /usr/local/go/bin/go mod download
ðŸ¦ - $ /usr/local/go/bin/go build -o /workspace/artifacts/plugins/linux/arm64/kubernetes/feature/v0.1.0/tanzu-feature-linux_arm64 -ldflags  -X 'github.com/vmware-tanzu/tanzu-plugin-runtime/plugin/buildinfo.Version=v0.1.0' -tags  ./.
ðŸ¦ - error: exit status 1
ðŸ¦ - output: error obtaining VCS status: exit status 128
 	Use -buildvcs=false to disable VCS stamping.
ðŸ¦ - error compiling plugin feature
```

To get passed this error, I need to pass in the goflag, `-buildvcs=false`.

**Describe the solution you'd like**
Enable passing in goflags for plugin builds. 

**Describe alternatives you've considered**
- change Docker context to include parent of submodule
- use Git Clone
- copy over git submodule's parent's .git file to submodule

**Additional context**

### PR DESCRIPTION
**Pull request by codegold79**
# Description
_Include a summary of the change. If this change fixes an issue, indicate that on the `Fixes` line. Briefly include
relevant context and motivation._

Fixes #90

This PR has commits that do the following:
- Restrict plugin builds to amd64 architectures 
- Find only directories and not other types of files in cmd/plugin
- Pass goflags to Tanzu CLI  plugin builder
- Pass in specific linter version, not latest

## Change Details
Check all that apply:
- [x] New feature 
- [x] Bug fix 
- [ ] Non-code change 
- [ ] Non-breaking change 
- [ ] Breaking change
- [ ] Requires a documentation change

## Release Note

```release-note
Pass goflags to Tanzu CLI Plugin Builder for plugin builds
```

# How Has This Been Tested?
_Describe any testing done to verify changes. Provide enough detail that tests can be reproduced._

I ran this in my team's downstream pipeline and it is able to successfully build feature and coregen plugins from the Tanzu Framework repo.

# Checklist:
- [x] My code follows the [Contribution and Style guidelines](../../CONTRIBUTING.md) of this project
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added test details that prove my fix is effective or that my feature works

### CODE REVIEW

File: templates/Dockerfile [hunk 1]
```diff
@@ -14,7 +15,7 @@ RUN --mount=target=. \
     cd $COMPONENT && go mod download
 
 # Linting
-FROM golangci/golangci-lint:latest AS lint-base
+FROM --platform=${BUILDPLATFORM} $LINTER_BASE_IMAGE AS lint-base
```
Comment (yharish991) [inline templates/Dockerfile:18]: lets's not let the user configure this image. On the downstream side, we can do the same as how we did for golang image

File: templates/Dockerfile [hunk 2]
```diff
@@ -84,11 +85,15 @@ FROM cli-plugin-builder-install AS cli-plugin-build-prep
 ARG CLI_PLUGIN_VERSION
 ARG CLI_PLUGIN
 ARG OCI_REGISTRY
+ARG CLI_PLUGIN_GO_FLAGS
 RUN --mount=type=bind,readwrite \
     --mount=from=carvel-base,src=/bin/imgpkg,target=/bin/imgpkg \
-    tanzu-builder plugin build --match "${CLI_PLUGIN}" \
+    tanzu-builder plugin build \
+        --match "${CLI_PLUGIN}" \
+        --os-arch linux_amd64 --os-arch windows_amd64 --os-arch darwin_amd64 \
```
Comment (yharish991) [inline templates/Dockerfile:93]: curious, what is this flag for?

File: templates/Dockerfile (hunk 1 above)
Comment (yharish991) [reply templates/Dockerfile:18]: i'm +1 with pinning the version though :)

File: templates/Dockerfile (hunk 2 above)
Comment (yharish991) [reply templates/Dockerfile:92]: let's check with cli team on which os-arch combinations we need to support for Tanzu cli and make changes accordingly..

Comment (codegold79) [reply templates/Dockerfile:92]: Yeah, but meanwhile, let's get PR merged. We can always change the os-arch combinations later.

**Review by yharish991 — APPROVED** [review]
LGTM

**Review by rajathagasthya — APPROVED** [review]

**Pull request merged**